LibSWBF2/Wrappers/Light.cpp

```diff
@@ -0,0 +1,175 @@
+#include "Light.h"
+
+#include <stdint.h>
+
+void read3Floats(const uint8_t *rawPtr, 
+				float_t& x, float_t& y,
+				float_t& z){
+    const float_t *floatPtr = reinterpret_cast<const float_t *>(rawPtr);
+    x = *(floatPtr); 
+    y = *(floatPtr + 1); 
+    z = *(floatPtr + 2);   
+}
+
+void read4Floats(const uint8_t *rawPtr, 
+				float_t& x, float_t& y,
+                float_t& z, float_t& w){
+    const float_t *floatPtr = reinterpret_cast<const float_t *>(rawPtr);
+    x = *(floatPtr); 
+    y = *(floatPtr + 1); 
+    z = *(floatPtr + 2);
+    w = *(floatPtr + 3);   
+}
```

Comment:
Ah shit, wish I could've done this review earlier...
I'm just writing the major stuff right here.

Your idea of using the `RawData` chunk and interpret from there in the wrapper is not bad, since the `DATA` chunk appears in several contexts. Downside here is, there is no benefit in holding the raw chunk data at this point. The idea of mirroring the chunk data in structs has two major upsides: Having access to the data fields as represented in the file, and the ability to manipulate data on that low level. The `RawData` approach fulfills the first one, but not the second.

In general, the wrapper classes just serve as an abstraction layer to more easily access desired data, but they're not a necessity. You could just go and access the chunks directly if you want.

For example, the `INFO` chunk appears in a lot of contexts and is different depending on it (e.g. an `INFO` chunk as a child of `tex_` is different than an `INFO` chunk as a child of `segm`). For each of those `INFO` cases, there is a different `INFO` classes (`tex_.INFO.h`, `segm.INFO.h`, ...), so there is no `Chunks/LVL/common/INFO.h`. although I must say I have not been consistent with the file naming everywhere... This is another TODO for me xD

Anyway, what I wanna say is that you should discard the `Chunks/LVL/common/DATA.h` approach and just go with an explicit implementation just for lights: `Chunks/LVL/lght/DATA.h` and reading all the fields right there.

Reply:
About the `read3Float` and `read4Floats` stuff, you don't need them anymore after doing an explicit `DATA` implementation. But yes, in that implementation you should just use `Vector3/4::ReadFromStream`.